Content.Server/GameObjects/Components/Chemistry/SolutionAreaEffectComponent.cs

```diff
@@ -0,0 +1,188 @@
+﻿#nullable enable
+using System;
+using System.Linq;
+using Content.Server.GameObjects.Components.Atmos;
+using Content.Server.Utility;
+using Content.Shared.Chemistry;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.ComponentDependencies;
+using Robust.Shared.GameObjects.Components.Transform;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Map;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Maths;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.GameObjects.Components.Chemistry
+{
+    /// <summary>
+    /// Used to clone its owner repeatedly and group up them all so they behave like one unit, that way you can have
+    /// effects that cover an area. Inherited by <see cref="SmokeSolutionAreaEffectComponent"/> and <see cref="FoamSolutionAreaEffectComponent"/>.
+    /// </summary>
+    public abstract class SolutionAreaEffectComponent : Component
+    {
+        [Dependency] protected readonly IMapManager MapManager = default!;
+        [Dependency] protected readonly IPrototypeManager PrototypeManager = default!;
+
+        [ComponentDependency] protected readonly SnapGridComponent? SnapGridComponent = default!;
+        [ComponentDependency] protected readonly SolutionContainerComponent? SolutionContainerComponent = default!;
+        public int Amount { get; set; }
+        public SolutionAreaEffectInceptionComponent? Inception { get; set; }
```

Comment:
the inceptioncomponent does not need to be an actual component. also the ticking should probably just be moved into this class.

What you should do is:
- Move `InceptionUpdate` into this class
- Replace the component with a `SolutionAreaEffectGroup`-Class that still has the add/remove functions. The newly ported over function then just takes the hashset from inside the solutionareaeffectgroup

Reply:
DONT. FORGET WHAT I SAID